draft-ietf-tls-tls13.md

```diff
@@ -4696,6 +4641,193 @@ a context in all exporter computations, though the value could be empty.
+
+The server MUST derive the storage key only from validated sections
+of the ClientHello. If the ClientHello contains multiple
+PSK identities, then an attacker can create multiple ClientHellos
+with different binder values for the less-preferred identity on the
+assumption that the server will not verify it. I.e., if the
+client sends PSKs A and B but the server prefers A, then the
+attacker can change the binder for B without affecting the binder
+for A. If the validated binder or the ClientHello.random
+are used for this purpose, then this attack is not possible.
+
+Because this mechanism does not require storing all outstanding
+tickets, it may be easier to implement in distributed systems with
+high rates of resumption and 0-RTT, at the cost of potentially
+weaker anti-replay defense because of the difficulty of reliably
+storing and retrieving the received ClientHello messages.
+In many such systems, it is impractical to have globally
+consistent storage of all the received ClientHellos.
+In this case, the stronger design is to have a single storage zone be
+authoritative for a given ticket and refuse 0-RTT for that
+ticket in any other zone. This approach prevents simple
+replay by the attacker because only one zone will accept
+0-RTT data. A weaker design is to implement separate storage for
+each zone but allow 0-RTT in any zone. This approach limits
+the number of replays to once per zone. Application message
+duplication of course remains possible with either design.
+
+When implementations are freshly started, they SHOULD
+reject 0-RTT as long as any portion of their recording window overlaps
+the startup time. Otherwise, they run the risk of accepting
+replays which were originally sent during that period.
+
+Note: If the client's clock is running much faster than the server's
+then a ClientHello may be received that is outside the window in the
+future, in which case it might be accepted for 1-RTT, causing a client retry,
+and then acceptable later for 0-RTT. This is another variant of
+the second form of attack described above.
+
+
+
+## Relevance Checks
+
+Because the ClientHello indicates the time at which the client sent
+it, it is possible to efficiently determine whether a ClientHello was
+likely sent reasonably recently and only accept 0-RTT for such
+ClientHello, otherwise falling back to a 1-RTT handshake.
+This is necessary for the ClientHello storage mechanism
+described in {{client-hello-recording}} because otherwise the server
+needs to store an unlimited number of ClientHellos and is a useful optimization for
+single-use tickets because it allows efficient rejection of ClientHellos
+which cannot be used for 0-RTT.
+
+In order to implement this mechanism, a server needs to store the time
+that the server generated the session ticket, offset by an estimate of
+the round trip time between client and server. I.e.,
+
+~~~~
+    adjusted_creation_time = creation_time + estimated_RTT
+~~~~
```

Comment:
I think this is really is the "estimated_max_RTT". If a client happens to have a 200ms RTT when the ticket is issued, that doesn't mean they'll show up with a 200ms RTT the next time. For Satellite and Mobile users,  the variance can be quite a lot; ranging between 50ms and 500ms isn't unusual on some networks. 

It would probably be too strict to reject 0-RTT just because the user got an unlucky connection. Especially since the slow connections benefit the most from 0-RTT.